src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -1086,7 +1086,7 @@ private static ulong DateToTicks(int year, int month, int day)
                 ThrowHelper.ThrowArgumentOutOfRange_BadYearMonthDay();
             }
 
-            ReadOnlySpan<uint> days = IsLeapYear(year) ? DaysToMonth366 : DaysToMonth365;
+            ReadOnlySpan<uint> days = month > 1 && IsLeapYear(year) ? DaysToMonth366 : DaysToMonth365;
```

Comment:
>month > 1 [](http://example.com/codeflow?start=38&length=9)

what is the benefit of this check? maybe worth a comment to the reader.

Reply:
I understand it is to avoid calling IsLeapYear. But wouldn't this check add checking cost when having other values of month?

Reply:
The main benefit is actually for the quite common case where a `DateTime` is constructed with const month=1,day=1 - this change allows skipping not only the leap year check but also the DaysToMonth lookup and check. The overhead for non-const cases is just one branch.

Reply:
> the quite common case where a DateTime is constructed with const month=1,day=1

I am not sure this is true. I rarely see DateTime constructed with that.